### PR TITLE
Test integration

### PR DESCRIPTION
**Pull request by jahandaniyal**

### CODE REVIEW

File: mockfog_application/tasks/check_dependencies.yml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+#  shell: sudo yum -y install python-pip
+#
+#- name: Install Docker
+#  shell: sudo amazon-linux-extras install docker -y
+#
+#- name: Add EMQX repo
+#  shell: sudo yum-config-manager --add-repo https://repos.emqx.io/emqx-ce/redhat/centos/7/emqx-ce.repo -y
+#
+#- name: Install EMQX
+#  shell: sudo yum install emqx -y
```
Comment (MaxF12) [inline mockfog_application/tasks/check_dependencies.yml]: Remove out commented code

File: mockfog_application/tasks/deploy.yml [hunk 2]
```diff
@@ -15,32 +49,82 @@
...
+    auto_remove: yes
+    log_driver: journald
+    exposed_ports: "{{ expose }}"
+    ports: "{{ ports }}"
+    env_file: "{{ folder }}/env_file.env"
+  when: app_vars.volumes is undefined or app_vars.volumes|length == 0
+  ignore_errors: true
+
+#- name: Start docker application containers
+#  shell: docker run -d -m256M --cpus 2 -it --rm -p {{ ports }} {{ docker_image }}
```
Comment (MaxF12) [inline mockfog_application/tasks/deploy.yml]: This too

File: testbed/common.py [hunk 3]
```diff
@@ -101,7 +101,14 @@ def resolve_names(g):
...
-        for config in attrs['app_configs']:
-            config['internal_ip'] = name_to_ip[config['connect_to']]
-        g.node[node]['app_configs'] = attrs['app_configs']
+    # for node, attrs in machine_nodes:
+    #     for config in attrs['app_configs']:
+    #         config['internal_ip'] = name_to_ip[config['connect_to']]
+    #         app_configs[node] = {}
+    #         app_configs[node].update(config)
+    #     # g.node[node]['app_configs'] = attrs['app_configs']
+    #     del g.node[node]['app_configs']
```
Comment (MaxF12) [inline testbed/common.py]: Same

File: testbed/common.py [hunk 4]
```diff
@@ -45,7 +45,7 @@ def app_config(**kwargs):
     config = {
         **kwargs,
     }
-    assert 'timeout' in config
+    # assert 'timeout' in config
```
Comment (MaxF12) [inline testbed/common.py]: +1

File: testbed/generate_testbed_definition.py [hunk 5]
```diff
@@ -6,14 +6,22 @@
...
+    resolve_names,
+    validate_graph
+)
 
 g = nx.Graph()
 
 # Generate topology
 
-topologies.simple_topology(g)
+# topologies.simple_topology(g)
```
Comment (MaxF12) [inline testbed/generate_testbed_definition.py]: +1

File: mapping.json [hunk 6]
```diff
@@ -0,0 +1,9 @@
+[
+{"ip": "52.29.200.79", "name": "generator2" },
```
Comment (renderfehler) [inline mapping.json]: What is this file for?

File: mapping.yml [hunk 7]
```diff
@@ -0,0 +1,14 @@
+- ip: 52.29.200.79
+  name: generator2
+- ip: 3.120.111.80
```
Comment (renderfehler) [inline mapping.yml]: What is this file for? Isn't this file generated each time?

File: mockfog_agent/telegraf.conf [hunk 8]
```diff
@@ -0,0 +1,11 @@
+[[outputs.influxdb]]
```
Comment (renderfehler) [inline mockfog_agent/telegraf.conf]: Is this still included? We don't need this anymore.

File: mockfog_application/tasks/check_dependencies.yml [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+    ports: "{{ item.ports }}"
+  with_items: "{{ _app_def }}"
+  when: item.name == "{{ app_def }}"
+
+#- name: Install boto3
+#  shell: sudo pip3 install boto3
+
+#- name: Install pip
+#  shell: sudo yum -y install python-pip
+#
```
Comment (renderfehler) [inline mockfog_application/tasks/check_dependencies.yml]: Remove the comments?

File: mockfog_application/tasks/deploy_telegraf.yml [hunk 10]
```diff
@@ -0,0 +1,19 @@
+---
```
Comment (renderfehler) [inline mockfog_application/tasks/deploy_telegraf.yml]: Do we want to keep this?

File: mockfog_application/tasks/main.yml [hunk 11]
```diff
@@ -1,7 +1,15 @@
...
 - import_tasks: collect.yml
   tags: collect
+
+- import_tasks: check_dependencies.yml
+  tags: check_dependencies
+
+- import_tasks: deploy_agent.yml
+  tags: deploy_agent
+
+- import_tasks: deploy_telegraf.yml
```
Comment (renderfehler) [inline mockfog_application/tasks/main.yml]: Here again deploy telegraf

File: setup.py [hunk 12]
```diff
@@ -0,0 +1,9 @@
+import os
```
Comment (renderfehler) [inline setup.py]: This should be gone too. The agent should be a single file now.

File: testbed/generate_topologies.py [hunk 13]
```diff
@@ -0,0 +1,33 @@
...
+        print(exc)
+
+
+def add_node(g: Graph, node):
+    # zones are used to prevent cyclies
+    if not node['role']:
+        g.add_node(node['name'], **node_attrs(type='zone'))
+
+    # if you do not supply a zone, the node is interpreted as being a machine
+    # the role is attached as tag to the AWS instance and can be used to run tasks only on machines with a certain role (see mockfog_application.yml notebook)
```
Comment (renderfehler) [inline testbed/generate_topologies.py]: indentation

File: testbed/testbed_definition.yml [hunk 14]
```diff
@@ -3,51 +3,182 @@ nodes:
 - type: machine
```
Comment (renderfehler) [inline testbed/testbed_definition.yml]: Add all generated files to .gitignore

File: testbed/application_definition.yml [hunk 15]
```diff
@@ -0,0 +1 @@
+--- {}
```
Comment (renderfehler) [inline testbed/application_definition.yml]: What is this for?

**Review by renderfehler — CHANGES_REQUESTED** [review]

**Review by renderfehler — APPROVED** [review]

**Pull request merged**